src/test/java/StringManipulationTest.java

```diff
@@ -26,58 +27,117 @@ public void testCount1() {
-        fail("Not yet implemented");
+
+        manipulatedstring.setString("How many words is this? It can count correctly, I hope.");
+        int length = manipulatedstring.count();
+        assertEquals(11, length);
     }
 
+    // This test checks if the method properly returns the number of words
+    // when the string contains a colon and a space.
     @Test
     public void testCount4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Item: 4");
+        int length = manipulatedstring.count();
+        assertEquals(2, length);
+    }
+
+    // This test checks if method returns zero when the manipulatedstring is set to
+    // an empty string
+    @Test
+    public void testCount5() {
+        manipulatedstring.setString("");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
     }
 
+    // This test checks if the method throws NullPointerException if the count() is called
+    // on the string that has been set to null
+    @Test
+    public void testCount6() {
+        manipulatedstring.setString(null);
+        assertThrows(NullPointerException.class, ()->{manipulatedstring.count();});
+    }
+
+
     @Test
     public void testRemoveNthCharacter1() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
         assertEquals("I' bttr uts0e 16tsinths trn6 rgh?", manipulatedstring.removeNthCharacter(3, false));
     }
 
+
     @Test
     public void testRemoveNthCharacter2() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
         assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
     }
 
+    // This test checks if the method throws NullPointerException if the string has been set to null.
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString(null);
+        assertThrows(NullPointerException.class, ()->{manipulatedstring.removeNthCharacter(1, true);});
     }
 
+    // This test checks if the method throws IllegalArgumentException if n is 0.
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I'm a CS student.");
+        assertThrows(IllegalArgumentException.class, () -> {manipulatedstring.removeNthCharacter(0, true);});
     }
-
+    //This test checks if the method throws IndexOutOfBoundsException if n is
+    // greater than the length of the string
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I live in Seattle.");
+        // nValue holds the length of the string plus one
+        int nValue = manipulatedstring.toString().length() + 1;
+        assertThrows(IndexOutOfBoundsException.class, () -> {manipulatedstring.removeNthCharacter(nValue, false);});
     }
```

Comment:
Good looking test:
Here's another approach you could take as well.
```
@Test
public void testRemoveNthCharacter5() {
    manipulatedstring.setString("I live in Seattle.");
    int nValue = manipulatedstring.toString().length() + 1;
    
    try {
        manipulatedstring.removeNthCharacter(nValue, false);
        fail("Expected IndexOutOfBoundsException to be thrown.");
    } catch (IndexOutOfBoundsException e) {
        // Asserting that the exception message contains relevant information
        assertTrue(e.getMessage().contains("Invalid character index"));
    }
}
```

Reply:
Thank you for telling me how to use a different approach. I will keep my test case as-is this time because it works as expected, but I will keep in mind that I could use an approach like this in the future:)